config/config.go

```diff
@@ -0,0 +1,519 @@
+//       FirstName string `desc:"Desc for First Name"`
+//       Age       uint16 `desc:"Desc for Age"`
+//   }
+// The struct has two fields (with prefix example):
+//   Field       Flag                   ENV                  Key (In config file)
+//   FirstName   --example-first-name   EXAMPLE_FIRST_NAME   example.firstName
+//   Age         --example-age          EXAMPLE_AGE          example.age
+// When you execute command with `--help`, you can see the help doc of flags and
+// descriptions (From field tag `desc`).
+//
+// The priority is:
+//   Flag > ENV > Key > The value you set in option
+type Option interface{}
+
+// Plugin is for plugins to collect configurations
+type Plugin interface {
+	// Name returns plugin name.
+	Name() string
+	// Configure configures nirvana config via current options.
+	Configure(cfg *nirvana.Config) error
+}
+
+// NirvanaOption contains basic configurations of nirvana.
+type NirvanaOption struct {
+	// IP is the IP to listen.
+	IP string `desc:"Nirvana server listening IP"`
+	// Port is the port to listen.
+	Port uint16 `desc:"Nirvana server listening Port"`
+}
+
+// NewDefaultNirvanaOption creates a default nirvana option.
+func NewDefaultNirvanaOption() NirvanaOption {
+	return NirvanaOption{
+		IP:   "",
+		Port: 80,
+	}
+}
+
+// Name returns plugin name.
+func (p *NirvanaOption) Name() string {
+	return "nirvana"
+}
+
+// Configure configures nirvana config via current options.
+func (p *NirvanaOption) Configure(cfg *nirvana.Config) error {
+	cfg.Configure(
+		nirvana.IP(p.IP),
+		nirvana.Port(p.Port),
+	)
+	return nil
+}
+
+// NirvanaCommand is a nirvana command.
+type NirvanaCommand interface {
+	// EnablePlugin enables plugins.
+	EnablePlugin(plugins ...Plugin) NirvanaCommand
+	// AddOption will fill up options from flags/ENV/config after executing.
+	// A non-empty prefix is recommended. It's used to divide option namespaces.
+	AddOption(prefix string, options ...Option) NirvanaCommand
+	// Add adds a field by key.
+	// If you don't have any struct to describe an option, you can use the method to
+	// add a single field into nirvana command.
+	// `pointer` must be a pointer to golang basic data type (e.g. *int, *string).
+	// `key` must a config key. It's like 'nirvana.ip' and 'myconfig.name.firstName'.
+	// The key will be converted to flag and env (e.g. --nirvana-ip and NIRVANA_IP).
+	// If you want a short flag for the field, you can only set a one-char string.
+	// `desc` describes the field.
+	Add(pointer interface{}, key string, shortFlag string, desc string) NirvanaCommand
+	// Execute runs nirvana server.
+	Execute(descriptors ...definition.Descriptor) error
+	// ExecuteWithConfig runs nirvana server from a custom config.
+	ExecuteWithConfig(cfg *nirvana.Config) error
+	// Command returns a command for command.
+	Command(cfg *nirvana.Config) *cobra.Command
+}
+
+// NewNirvanaCommand creates a nirvana command.
+func NewNirvanaCommand(option NirvanaOption) NirvanaCommand {
```

Comment:
Can you use pointer here, so it will be consistent with passing options to plugins.